libs/payments/paypal/src/lib/paypal.service.ts

```diff
@@ -4,8 +4,48 @@
 
 import { Injectable } from '@nestjs/common';
 import { PayPalManager } from './paypal.manager';
+import { PaypalCustomerManager } from './paypalCustomer/paypalCustomer.manager';
+import { CreatePaypalCustomer } from './paypalCustomer/paypalCustomer.types';
+import { CurrencyManager } from '@fxa/payments/currency';
 
 @Injectable()
 export class PayPalService {
-  constructor(private paypalManager: PayPalManager) {}
+  constructor(
+    private paypalManager: PayPalManager,
+    private paypalCustomerManager: PaypalCustomerManager,
+    private currencyManager: CurrencyManager
+  ) {}
+
+  /**
+   * Create and verify a billing agreement is funded from the appropriate
+   * country given the currency of the billing agreement.
+   */
+  async createBillingAgreement(options: {
+    uid: string;
+    token: string;
+    currency: string;
+  }) {
```

Comment:
Since none of these are optional, these should each be independent arguments.

Reply:
Updated!